src/workerd/io/io-context.h

```diff
@@ -428,6 +428,12 @@ class IoContext final: public kj::Refcounted, private kj::TaskSet::ErrorHandler
   bool isCurrent();
   // True if this is the IoContext for the current thread (same as `hasCurrent() && tcx == current()`).
 
+  struct SameIoContextScope {
```

Comment:
Why a scope object rather than a method? Something like `context.requireCurrent()` seems like it would be equivalent?

Reply:
> Why a scope object rather than a method?

Honestly just a personal style preference and habit.

Reply:
Hmm, personally I find it confusing. I would expect a scope object actually protects a scope (which implies having a meaningful destructor) rather than being a point check.

Reply:
ok, updated. Replaced with a `ioContext.requireCurrentOrThrowJs()` method.